pickhardtpayments/Attempt.py

```diff
@@ -0,0 +1,134 @@
+from enum import Enum
+
+from pickhardtpayments import Channel
+
+
+class SettlementStatus(Enum):
+    PLANNED = 1
+    INFLIGHT = 2
+    ARRIVED = 3
+    FAILED = 4
+    SETTLED = 5
+
+
+class Attempt:
+    """
+    An Attempt describes a path (a set of channels) of an amount from sender to receiver.
+
+    When sending an amount of sats from sender to receiver, a payment is usually split up and sent across
+    several paths, to increase the probability of being successfully delivered. Each of this path is referred to as an
+    Attempt.
+    An Attempt consists of a list of Channels (class:Channel) and the amount in sats to be sent through this path.
+
+    :param path: a list of UncertaintyChannel objects from sender to receiver
+    :type path: list[UncertaintyChannel]
+    :param amount: the amount to be transferred from source to destination
+    :type amount: int
+    """
+
+    def __init__(self, path: list[Channel], amount: int = 0):
+        """Constructor method
+        """
+        self._paths = None
+        self._routing_fee = -1
+        self._probability = -1
+        if not isinstance(path, list):
+            raise ValueError("path needs to be a collection of Channels")
+        for channel in path:
+            if not isinstance(channel, Channel):
+                raise ValueError("path needs to be a collection of Channels")
+            self._path = path
+        self._status = SettlementStatus.PLANNED
+        self._amount = amount
+
+    def __str__(self):
+        return "Path with {} channels to deliver {} sats and status {}.".format(len(self._path),
+                                                                                self._amount, self._status.name)
+
+    @property
+    def path(self):
+        """Returns the path of the attempt.
+
+        :return: the list of UncertaintyChannels that the path consists of
+        :rtype: list[UncertaintyChannel]
+        """
+        return self._path
+
+    @path.setter
+    def path(self, attempts: list):
+        """Sets the path that was set up for a certain amount
+
+        :param attempts: a List of Channels from UncertaintyGraph that an amount was attempted to be routed through
+        :type attempts: list[UncertaintyChannel]
+        """
+        if not isinstance(attempts, list):
+            raise ValueError("path needs to be a collection of Channels")
+        for attempt in attempts:
+            if not isinstance(attempt, Channel):
+                raise ValueError("path needs to be a collection of Channels")
+        self._paths = attempts
```

Comment:
I really don't get this :(

Reply:
this is indeed unnecessary as well. 
This is part of the setter for path in Attempt. When I originally added the path to attempt, I used this setter. And while I refactored step by step, at the time i wasn't sure that didn't make any mistakes when passing on parameters. That's why i checked before inclusion and raised errors to see where things failed. 

The setter now can go, because I made this a mandatory parameter at instantiation and I require the correct class. 

Thanks for pointing this out!